src/main/kotlin/no/nav/syfo/kafka/consumers/syketilfelle/SyketilfelleKafkaConsumer.kt

```diff
@@ -32,25 +32,33 @@ class SyketilfelleKafkaConsumer(
     }
 
     override suspend fun listen(applicationState: ApplicationState) {
-        log.info("Started listening to topic $topicFlexSyketilfellebiter")
+        LOG.info("Started listening to topic $topicFlexSyketilfellebiter")
         while (applicationState.running) {
             kafkaListener.poll(zeroMillis).forEach {
                 try {
                     val kSyketilfellebit: KSyketilfellebit = objectMapper.readValue(it.value())
                     databaseInterface.storeSyketilfellebit(kSyketilfellebit.toPSyketilfellebit())
                     val sykmeldtFnr = kSyketilfellebit.fnr
-                    if (accessControl.canUserBeNotified(sykmeldtFnr) && kSyketilfellebit.orgnummer != null) {
+                    val userAccessStatus = accessControlService.getUserAccessStatusByFnr(sykmeldtFnr)
+
+                    if (kSyketilfellebit.orgnummer != null) {
                         varselPlanners.forEach {
-                            it.processSyketilfelle(sykmeldtFnr, kSyketilfellebit.orgnummer)
+                            varselPlanners.forEach { planner ->
+                                if (planner.varselSkalLagres(userAccessStatus)) {
```

Comment:
Jeg synes dette kan sjekkes av hver enkelt planner. Ellers blir logikk for om man planlegger et varsel eller ikke distribuert, og det blir vanskeligere å skjønne hvorfor man har fått eller ikke fått planlagt et varsel.

Reply:
Da slipper man også å innføre et nytt interface

Reply:
Det hadde jeg i utganspunktet, men flyttet til interface for å unngå duplisering da vi har `MER_VEILEDNING` plannere som er kalt i kontekst av `SyketilfelleKafkaConsumer` og `OppfolgingstilfelleKafkaConsumer` og logikkmessig er helt like.

Reply:
Ikke uenig med deg, hadde det slikt i noen iterasjoner, men foretrukket dette. Man kan jo passere `userAccesStatus `som argument i `processOppfolgingstilfelle() `og i `processSyketilfelle()` , men da blir det mange duplikater, og akkurat tilganger-greier har ikke noe å gjøre med forretningslogikk for beregninger som ligger i planner-klassene. I tillegg synes at det er lurt å sjekke tilganger tidlig og på ett sted. 
(hvis jeg tolket det du skrev riktig 🙌🏻 )
Ikke sånn at jeg har super sterk mening om det, men tror at det er litt cleanere med ekstra interface.